client/post-editor/index.js

```diff
@@ -14,6 +14,7 @@ module.exports = function() {
 	page( '/post', controller.pressThis, sitesController.siteSelection, sitesController.sites );
 	page( '/post/new', () => page.redirect( '/post' ) ); // redirect from beep-beep-boop
 	page( '/post/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.post );
+	page( '/post/revision/:site?/:post?', sitesController.siteSelection, sitesController.fetchJetpackSettings, controller.revision );
```

Comment:
> Aren't we just adding a 3rd position here? So, the only changes to getSiteFragment needed to support /post/:site/:post/revisions should be as follows:

The more positions we accept, the harder it will be to maintain and the more chances we have of breaking something. After having read #9561, #9445 and #4614 it definitely feels like this is already quite fragile, hence why I'm kinda reluctant at touching it :) 

I proposed the idea of using `page.js` params as that would free us from any positional guessing. The probability of having a slug-like (or number) at a position where we interpret it as a site identifier is considerably higher than naming a param `:site` or `:siteId` that wouldn't actually contain a site identifier. The (hopefully) net gain in maintainability makes the "risk" of touching it potentially worth it (but it's still a (very) risky move imho)